src/HonzaBotner.Discord.Services/Attributes/OnlyForMemberCountAllowlistedAttribute.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using DSharpPlus.CommandsNext;
+using DSharpPlus.CommandsNext.Attributes;
+using DSharpPlus.Entities;
+using HonzaBotner.Discord.Attributes;
+using HonzaBotner.Discord.Services.Options;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace HonzaBotner.Discord.Services.Attributes;
+
+/// <summary>
+/// Defines that usage of this command is only allowed to:
+/// - moderators
+/// - or members that are allow-listed (in config) to do so.
+/// </summary>
+public class OnlyForMemberCountAllowlistedAttribute : CheckBaseAttribute, IOnlyForMemberCountAllowlistedAttribute
```

Comment:
I strongly disagree with the name here. From what was named "RequireTeacher" and was only in one class with potential to be universal, we moved to "OnlyForMemberCountAllowlistedAttribute" which is now universal even though name suggests only one use scenario - thus disallowing meaningfull usage elsewhere even though it has a potential to do so

Reply:
Attributes cannot be (as I understand) in a disjunct relationship, ergo if we want to allow mods + all allowlisted roles (for now its Teacher, but we should keep code future-wise), we cannot use some generic naming and functionality.

Reply:
Some superb ideas on how to solve this are welcomed. :)

Reply:
Ok, regarding this....
### If I start with the name
In the first commit which I was originaly reacting to, you've decided to introduce `TeacherRoleId` config value, which was imo totally awesome change, in line with idea behind this bot - We are providing services on a *school/student* server and it might be useful to know which role is teacher and which is student. - On the other side, you have decided to use this config value only on one place in a non-universal way, so I proposed "Hey, you are using something totaly awesome like checking for teacher role, that could be used elsewhere, but you use it only on one place!!! Make it universal". I mean, it is a school bot and we might run into more scenarios where it is needed.

The changes you implemented are what I don't like. You've decided to take something with potential to be universal and turn it into one time one place use. I mean, where am I going to use `OnlyForMemberCountAllowlistedAttribute`? I see, maybe when I'm working with Member count command? But.... where else? Why is it attribute if I can use it only on one place? Either you can rename the config value thus totaly screwing the universality, but keeping it local to that class, or you can move that requirement check and make it attribute, but make it usable also elsewhere. You can't do both.
### Regarding permissions
Imo when there is some system of permissions anywhere, giving rights to use something to the lowest role gives access also to the role higher. If we assume that roles on our server in terms of access to bot commands are `Pleb` -> `Student` -> `Teacher` -> `Mod`, then giving rights to teacher means that mod can use it too. Which is what you went for here (at least before when it was part of the class). And imo is totally fine. If this was `RequireTeacherAttribute` then it would make perfect sense that there is check for Mod role too. If we have some commands that require authentication (= role that says "Hey, I'm student!"), then it makes perfect sense that teachers and mods can use it too.
### Proposed solution
I am rooting for `RequireTeacherAttribute`, but if you are against it and want approach which gives you more freedom when picking roles, then I'm for this:
- Change the config value to something like `CommandSpecificPermissions`. It's value will be dictionary of strings and lists of ints, so something like this:
```
CommandSpecificPermissions: 
   {
   MemberCount: [roleID1, roleID2, roleID3],
   VeryNiceCommand: [roleID1, roleID2, roleID3],
   YouGetTheIdea.....
   }
```
Then, when doing the attribute check, you pass in the value. So let's say it's called 'RequireSetOfRolesAttribute' (It's a terible name), the you would do something like this:
```
[Description("Counts members by provided roles.")]
    [RequireSetOfRolesAttribute("MemberCount")]
    [ModuleLifespan(ModuleLifespan.Transient)]
    public class MemberRoleCount : BaseCommandModule
```
and inside the method, you would check for all the IDs in the dictionary on a key with that value that you passed in.